components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/DefaultOAuth2TokenValidator.java

```diff
@@ -42,38 +52,82 @@ public boolean validateAccessDelegation(OAuth2TokenValidationMessageContext mess
         return true;
     }
 
+    /**
+     *  Validate scope of the access token using scope validators registered for that specific app.
+     *
+     * @param messageContext Message context of the token validation request
+     * @return Whether validation success or not
+     * @throws IdentityOAuth2Exception Exception during while validation
+     */
     @Override
-    public boolean validateScope(OAuth2TokenValidationMessageContext messageContext)
-            throws IdentityOAuth2Exception {
+    public boolean validateScope(OAuth2TokenValidationMessageContext messageContext) throws IdentityOAuth2Exception {
+
+        String[] scopeValidators;
+        AccessTokenDO accessTokenDO = (AccessTokenDO) messageContext.getProperty(ACCESS_TOKEN_DO);
+        OAuthAppDO app;
+        try {
+            app = OAuth2Util.getAppInformationByClientId(accessTokenDO.getConsumerKey());
+            scopeValidators = app.getScopeValidators();
+        } catch (InvalidOAuthClientException e) {
+            throw new IdentityOAuth2Exception(String.format("Exception occurred when getting app information for " +
+                    "client id %s ", accessTokenDO.getConsumerKey()), e);
+        }
 
+        if (scopeValidators == null || scopeValidators.length == 0) {
```

Comment:
You can use ArrayUtils.isEmpty()

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/942/commits/ca4690d734709b108ef2df33a69ae3c9242febc8